File: Java/Algorithms and Data Structures/String Algorithms/ReverseInteger.java
public class ReverseInteger {
    public static int reverseInteger(int args) {
        int reverseOut = 0,input = args,rem = 0;
        while(input > 0){
            rem = input % 10;
            input = input / 10;
            reverseOut = (reverseOut * 10) + rem;
        }
        System.out.println(reverseOut);
        return reverseOut;
    }
    public static void main(String[] args) {
        int input = 89;
        reverseInteger(input);
    }
}

File: Python/Algorithms and Data Structures/String Algorithms/ReverseInteger.py

def checkConstraints(num): 
    if num == 0:
        return 0
    if num < -65536 or num > 65536:
        return 0
    elif num < 0 or num > 0:
        reverseInteger(num)
    else:
        None

def reverseInteger(num): #34
    reverse_integer = 0
    rem = 0
    if num > 0:
        while num > 0: 
            rem = num % 10 # 4 
            reverse_integer = reverse_integer * 10 + rem 
            num = (int)(num / 10) # 3
    else:
        num = num * (-1)
        while num > 0: 
            rem = num % 10 # 4 
            reverse_integer = reverse_integer * 10 + rem 
            num = (int)(num / 10) # 3
        reverse_integer = (-1) * reverse_integer
    print(reverse_integer)

num = -45
checkConstraints(num)


